GitHub-Test-4/mccolm-GitTest4/PythonNLPEx3.py

```diff
@@ -0,0 +1,98 @@
+import spacy
+import os
+import shutil
+from dicttoxml import dicttoxml
+# Silence Pandas deprecation warning (#54466)
+import pyarrow
+import pandas as pd
+import json
+
+nlp = spacy.load('en_core_web_md')
+
+
+def read_text_files(filepath, word: str):
+    with open(filepath, 'r', encoding='utf8') as f:
+        read_file = f.read()
+        string_file = str(read_file)
+
+        tokens = nlp(string_file)
+
+        word_of_interest = nlp(word)
+
+        high_similarity_dict = {}
+        for token in tokens:
+            if token and token.vector_norm:
+                if word_of_interest.similarity(token) > .3:
+                    high_similarity_dict[token] = word_of_interest.similarity(token)
+
+        high_similarity_reduced = {}
+        for key, value in high_similarity_dict.items():
+            if value not in high_similarity_reduced.values():
+                high_similarity_reduced[key] = value
+
+        sorted_sim_values = sorted(high_similarity_reduced.items(), key=lambda x: x[1], reverse=True)
+        return dict(sorted_sim_values)
+
+
+def output_file_json(dir_name: str, file_name: str, similarity_data: dict):
+    string_keys = {str(key): val for key, val in similarity_data.items()}
+    with open(f'{dir_name}/{file_name}.json', 'w') as file_handle:
+        json.dump(string_keys, file_handle)
+
+
+def output_file_csv(dir_name: str, file_name: str, similarity_data: dict):
+    df = pd.DataFrame.from_dict(similarity_data.items(), orient="columns")
+    df.columns = ['token', 'similarity']
+    df.to_csv(f'{dir_name}/{file_name}.tsv', sep='\t', index=False, encoding='utf-8')
```

Comment:
@MystKitteh I have a question about this: usually, the `'utf8'` is written without a hyphen. If you put in a hyphen, does it work the same way as if you do it without?